### DISCUSSION TITLE
More buttons for reporting packages that violate AUP

### DISCUSSION THREAD

**Discussion by AvnerCohen** [npm/feedback#492] [Website] [answered]
## The Context
This version of lang-list (https://www.npmjs.com/package/lang-list)
https://github.com/shadiabuhilal/lang-list/commit/2943c6b66ed244d03004685945b8a27e9d6c7213
Basically removes Hebrew and Yidish (intentionally).

This package is later on used in - https://www.npmjs.com/package/rtl-detect
And you can see in this ticket - https://github.com/shadiabuhilal/rtl-detect/issues/11
That this package is no longer working for Hebrew and Yidish.

## Discussion

I suspect that this could be a slippery slope.
A war over npm packages between different political agendas (imagine a package that fails when installed in china or russia based on IP range, or one that fails if the word "black" is used in it).

At the end of the day, npm has the power to control this and control package distribution. This is the one gateway that can clearly block all political agendas that results in discrimination of specific groups of people.

## Suggestion 
Right now npm allows for "Report as Malware" for some of the package. I suggest to also add a "Report violation of COC" because this is clearly such example as per https://www.npmjs.com/policies/conduct

**Comment by MylesBorins** [comment, marked as answer]
Hey @AvnerCohen thank you for opening this discussion.

If you do believe that a project is in violation of our code of conduct the appropriate action is to open a support ticket at https://npmjs.com/support as outlined in our [code of conduct reporting instructions](https://www.npmjs.com/policies/conduct#reporting-violations-of-this-code-of-conduct).

I don't believe that public discussion about the validity of a Code of Conduct complaint is appropriate or fair so I hope that you can understand that I will not be speaking directly to the situation you outlined, but I do assure you that if you file a formal complaint via support that we will get back to you.

It seems like the actionable part of this discussion is specifically making it easier to report a code of conduct violation directly through the UI of a package. I'm not 100% that this would be the best idea. There are many reasons beyond malware that someone may wish to report a violation, many of them are listed in our [acceptable use policy](https://www.npmjs.com/policies/open-source-terms#acceptable-use) which includes violating the Code of Conduct.

Malware is particularly important and get routed in a highly escalated way so we can act quickly, we need to make sure that folks are not being compromised. We could make a button for "Report Violation of Acceptable Use Policy"... but this is both verbose, and may not really make sense to folks without extra context.

Alternatively to a button we could add a bit of text below the `report malware` button that says something akin to

> If you believe this module violates any of the terms in our [acceptable use policy](https://www.npmjs.com/policies/open-source-terms#acceptable-use) please open a ticket with [support](https://www.npmjs.com/support)

Thoughts?

**Reply by AvnerCohen** [reply]
Thanks @MylesBorins for the thoughtful repose. I have indeed created a support ticket as well, and as you correctly pointed out, the scope of this suggestion is to make it easier to report such situations using the website. Much like the "Report Malware case".

I tend to agree with your suggestion that malware type reports should be highest priority given potential impact.
That being said, I think that having a CoC violation reporting be more accessible, is an important statement just as well.

Given that the existing "Report Malware" button is technically just a pre-filled form to the same support ticket, I believe such   solution for CoC violations should be pretty straightforward.
Something like that:
![image](https://user-images.githubusercontent.com/1297254/125833974-21917411-364e-47b7-a87a-92845104512b.png)

**Reply by MylesBorins** [reply]
Thanks @AvnerCohen for taking the time to make this mock-up, helps a bunch.

There is definitely desire to improve this, but we need to discuss with a couple different teams about how to best do this and which options we should expose. I'm going to kick off some internal discussions and will aim to get back to you about it in the near future, but apologies in advance if this takes a bit of time to close the loop on.

**Reply by MylesBorins** [reply]
@AvnerCohen we have met with our Trust+Safety team and have an idea for how to improve this experience. Thanks for reporting
